lib/kitchen/provisioner/chef/policyfile.rb

```diff
@@ -69,20 +69,20 @@ def resolve
             run_command("#{cli_path} export #{escape_path(policyfile)} #{escape_path(path)} --force --chef-license #{license}")
           end
         end
-
+        
         # Runs `chef install` to determine the correct cookbook set and
         # generate the policyfile lock.
         def compile
-          if File.exist?(lockfile)
-            info("Installing cookbooks for Policyfile #{policyfile} using `#{cli_path} install`")
+          if File.exist?(lockfile) && always_update
+            info("Policy lock file exists and always_update is set, running `#{cli_path} update` for Policyfile #{policyfile}...")
+            run_command("chef update #{escape_path(policyfile)}")
           else
             info("Policy lock file doesn't exist, running `#{cli_path} install` for Policyfile #{policyfile}...")
-          end
-          run_command("#{cli_path} install #{escape_path(policyfile)} --chef-license #{license}")
-
-          if always_update
-            info("Updating policy lock using `#{cli_path} update`")
-            run_command("#{cli_path} update #{escape_path(policyfile)} --chef-license #{license}")
+            run_command("chef install #{escape_path(policyfile)}")
+            if always_update
+              info("Updating policy lock using `#{cli_path} update`")
+              run_command("chef update #{escape_path(policyfile)}")
+            end
```

Comment:
Is this even needed? if a lockfile doesn't exist then chef-install will pull all of the current deps and thus update would be redundant. Subsequent kitchen-converges would be caught in the first if statement with && always_update enabled and then would run update as needed.

Reply:
you are right - will remove, thanks